### ISSUE TITLE
Zero AUROC when validation set is only comprised of positive labels

### ISSUE THREAD

**Issue by LucaCappelletti94** [tensorflow/tensorflow#41081]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Tested on Ubuntu 18.04 and MacOS Catalina 10.15.5
- TensorFlow installed from (source or binary): installed from Conda (tensorflow-gpu) on Ubuntu and from pip on MacOS
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.7.6 (on both machines)
- CUDA/cuDNN version: 11.0 (on Ubuntu machine)
- GPU model and memory: 4 NVIDIA V100 GPUs

**Describe the current behavior**
The AUROC is zero when there are no negatives in the validation set. This may lead users to believe that the model is not performing correctly, while it is just an error related to a misuse of the AUROC metric.

**Describe the expected behavior**
The AUROC should be NaN when no negatives are present in the validation set. This way the user is aware that there is something wrong with the metric that is being used, with the possibility to add a warning since there is a 0/0 somewhere in the computation of the AUROC.

**Standalone code to reproduce the issue**
```python
from tensorflow.keras.metrics import AUC
metric = AUC()
metric.update_state([1, 1, 1], [1, 0.5, 0.3])
metric.result()
```

I believe that similar issues also happen for other metrics, as uncanny values were displayed also for Recall and Precision, but I have not identified a simple reproducible example.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.2, TF v2.3.0rc0 and TF-nightly (i.e. v2.4.0-dev20200705). Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/ffbc2e4b0e234b2e54122693e6cd4439/41081.ipynb). Thanks!

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue in TF 2.5. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/e722f9dff30466812d7138f4d38fe83e/copy-of-41081.ipynb) for reference.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

**Comment by LucaCappelletti94** [comment]
Here you go - the bug is still there.

Is version 2.17 recent enough? It is the default one on COLAB.

<img width="510" alt="Screenshot 2024-08-03 alle 07 56 48" src="https://github.com/user-attachments/assets/cc39c0c0-a8d3-42b1-b790-bbe09dd2a583">

**Comment by Kayyuri** [comment]
Hi @LucaCappelletti94,

Thank you so much for filing this issue! We truly appreciate your support. At the moment, this request is not prioritized and mark it for community contributions. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.
To avoid duplicate work, please leave a comment here if you decide to take it on!

Thank you again for your interest and we look forward to seeing your contributions!
